src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,163 @@
+package duke;
+
+// import java packages
+import java.util.ArrayList;
+
+// import duke packages
+import duke.command.Deadline;
+import duke.command.Event;
+import duke.command.Todo;
+import duke.task.Task;
+import duke.task.TaskList;
+import duke.util.Parser;
+import duke.util.Storage;
+import duke.util.Ui;
+
+/**
+ * A chatbot that stores tasks given by a user.
+ */
+public class Duke {
+    private Storage storage;
+    private TaskList commandList;
+    private Ui ui;
+
+    /**
+     * Creates a Duke chatbot.
+     *
+     * @param filePath Filepath to save tasks and load tasks.
+     */
+    public Duke(String filePath) {
+        ui = new Ui("", "");
+        storage = new Storage(filePath);
+        try {
+            commandList = new TaskList(storage.loadTasks());
+        } catch (DukeException e) {
+            ui.showLoadingError();
+            commandList = new TaskList();
+        }
+    }
+
+    /**
+     * Converts Duke to string format.
+     *
+     * @return Duke as a string.
+     */
+    @Override
+    public String toString() {
+        return "im a chatbot ☺ - calico";
+    }
+
+    /**
+     * Generates a response to user input.
+     *
+     * @param cmd Command from user
+     * @return Chatbot response
+     */
+    public String getResponse(String cmd) {
```

Comment:
For better code readability, you could avoid arrowhead style code with deep nesting and long functions! (From week 5 topics) Perhaps you could abstract out some of the functionality into a different method to make it shorter.